apps/webapp/app/routes/api.v1.runs.$runId.tasks.$id.callback.$secret.ts

```diff
@@ -0,0 +1,138 @@
+import type { ActionArgs } from "@remix-run/server-runtime";
+import { json } from "@remix-run/server-runtime";
+import type { CallbackTaskBodyOutput } from "@trigger.dev/core";
+import { CallbackTaskBodyInputSchema } from "@trigger.dev/core";
+import { RuntimeEnvironmentType } from "@trigger.dev/database";
+import { z } from "zod";
+import { $transaction, PrismaClient, PrismaClientOrTransaction, prisma } from "~/db.server";
+import { enqueueRunExecutionV2 } from "~/models/jobRunExecution.server";
+import { logger } from "~/services/logger.server";
+
+const ParamsSchema = z.object({
+  runId: z.string(),
+  id: z.string(),
+  secret: z.string(),
+});
+
+export async function action({ request, params }: ActionArgs) {
+  // Ensure this is a POST request
+  if (request.method.toUpperCase() !== "POST") {
+    return { status: 405, body: "Method Not Allowed" };
+  }
+
+  const { runId, id } = ParamsSchema.parse(params);
+
+  // Now parse the request body
+  const anyBody = await request.json();
+
+  // Allows any valid object
+  // TODO: maybe add proper schema parsing during io.runTask(), or even skip this step
+  const body = CallbackTaskBodyInputSchema.safeParse(anyBody);
```

Comment:
Currently `z.object({}).passthrough()`. Leave as is? Change to `z.any()` or even remove schema parsing? There are no user-provided schemas for remote callbacks (yet), and we couldn't use them here.

Reply:
I think we can probably skip the schema parsing and just save the body to the task output without it